src/urlencoded.rs

```diff
@@ -30,20 +30,20 @@ impl Params {
 }
 
 pub fn parse(encoded_string : &str) -> Params {
-    Params(form_urlencoded::parse(encoded_string.as_bytes()).into_iter().group())
+    Params(form_urlencoded::parse(encoded_string.as_bytes()).into_owned().group())
 }
 
 pub fn parse_uri(origin: &RequestUri) -> Params {
-    let f = |query: Option<&String>| query.map(|q| parse(&*q));
+    let f = |query: Option<&str>| query.map(|q| parse(&*q));
 
     let result = match *origin {
-        AbsoluteUri(ref url) => f(url.query.as_ref()),
-        AbsolutePath(ref s) => UrlParser::new().parse_path(&*s)
-                                                // FIXME: If this fails to parse,
-                                                // then it really shouldn't have
-                                                // reached here.
-                                               .ok()
-                                               .and_then(|(_, query, _)| f(query.as_ref())),
+        AbsoluteUri(ref url) => f(url.query()),
+        AbsolutePath(ref s) => {
+            let base = "f://".parse().unwrap();
```

Comment:
rust-url 1.0 is almost a rewrite. I removed a lot of code after changing the data structure in order to get anything to compile, and added things back / rewrote them gradually. I neglected to add standalone path parsing back in 1.0 or 1.1.

We could add it back in 1.2, but it’s not clear to me that that you want rust-url’s exact parser for this. The WHATWG URL spec is primarily designed for clients (typically, parsing a `<a href="...">` HTML attribute). Does an HTTP server want to every edge case the same way? For example, percent-encoding (not decoding) some punctuation/delimiters, normalizing `\` to `/`, normalizing `.` and `..` path segments?